### PR TITLE
fix(ssr): remove backref serialization

### PR DESCRIPTION
**Pull request by wmertens**
We don't need it

### CODE REVIEW

File: packages/qwik/src/core/tests/component.spec.tsx [hunk 1]
```diff
@@ -242,19 +242,19 @@ describe.each([
           <Component ssr-required>
             <div>
               <span>Component 1</span>
-              <Signal ssr-required>1</Signal>
```
Comment (Varixo) [inline packages/qwik/src/core/tests/component.spec.tsx:245]: this is wrong, it should be serialized, or we will create it every time we diff

File: packages/qwik/src/core/shared/serdes/serialize.ts [hunk 2]
```diff
@@ -510,22 +509,12 @@ export class Serializer {
...
-            // add all back refs to the roots
-            for (let i = 0; i < childVNodeData.length; i++) {
-              const value = childVNodeData[i];
-              if (isSsrAttrs(value)) {
-                const backRefs = tryGetBackRefs(value);
-                if (backRefs) {
-                  this.$serializationContext$.$addRoot$(backRefs);
-                }
-              }
-            }
```
Comment (Varixo) [inline packages/qwik/src/core/shared/serdes/serialize.ts:528]: This was correct, the thing to remove is ensureContainsBackRef for ssr

File: packages/qwik/src/core/tests/component.spec.tsx (hunk 1 above)
Comment (wmertens) [reply packages/qwik/src/core/tests/component.spec.tsx:245]: but it's not reactive, so this is an edge case. If it were reactive, it would ssr as a signal and include the data in the state.

I think it's better to create it on first diff than to send the state.

File: packages/qwik/src/core/shared/serdes/serialize.ts (hunk 2 above)
Comment (wmertens) [reply packages/qwik/src/core/shared/serdes/serialize.ts:528]: I don't understand why this is correct. Events are the only entry points to the app, and only the state they can access should be serialized, and everything useful will be discovered from forward state discovery.

File: packages/qwik/src/core/tests/component.spec.tsx (hunk 1 above)
Comment (Varixo) [reply packages/qwik/src/core/tests/component.spec.tsx:245]: I dont think so, if you have input inside non reactive virtual then it will reset. It has to stay
